File: php/src/Stack/Stack.php
<?php

namespace Claudiosilvajunior\Php\Stack;

class Stack
{
    private array $stack = [];
    private int $length = 0;

    public function empty(): bool
    {
        return empty($this->stack);
    }

    public function length(): int
    {
        return $this->length;
    }

    public function push(mixed $value): void
    {
        $this->stack[] = $value;
        $this->length++;
    }

    public function pop(): void
    {
        if ($this->empty()) {
            return;
        }

        array_pop($this->stack);
        $this->length--;
    }

    public function top(): mixed
    {
        return $this->stack[$this->length - 1] ?? null;
    }
}

File: java/src/main/java/org/example/stack/Stack.java
package org.example.stack;

import java.util.ArrayList;
import java.util.List;

public class Stack {
    private List<String> stack;
    private int length = 0;

    public Stack() {
        stack = new ArrayList<>();
    }

    public boolean empty() {
        return stack.isEmpty();
    }

    public void push(String value) {
        stack.add(value);
        length++;
    }

    public int length() {
        return length;
    }

    public void pop() {
        if (empty()) return;

        stack.remove(position());
        length--;
    }

    public String top() {
        return stack.get(position());
    }

    public int position() {
        return length > 0 ? length - 1 : length;
    }
}
